add content

mux99 opened the pull request:
this is an internal PR with the goal of validating and improving the contents of mptcp.dev before integrating to the remote

mux99 committed:
add content

mux99 commented:
PR now ready for review, the content can be previewed here:
https://mux99.github.io/mptcp.dev/

mux99 committed:
fix multiple fixes

- nav logo properly placed
- socket now colored
- am i using mptcp language
- improve this page link

mux99 committed:
test add dummy graph

mux99 committed:
fix highlight drop-down

mux99 committed:
fix liquid highlight compatibility

mux99 committed:
fix text colors

mux99 committed:
fix title multi-line

matttbe commented on _config.yml:
```diff
@@ -18,7 +18,7 @@
 # in the templates via {{ site.myvariable }}.
 
-title: Multipath TCP for Linux
+title: Multipath TCP<br>for Linux
```
the HTML title is now `<title>Home | Multipath TCPfor Linux</title>`
also, not related but should you not use `<br />` instead?

mux99 replied:
fixed

mux99 committed:
add empty pages

mux99 committed:
fix simpler details

mux99 committed:
[guide] add more content

mux99 committed:
fix fix title multi-line

mux99 committed:
[installation]

- add content to installation page
- moved routing config from implementation to installation

mux99 committed:
add ss source

mux99 committed:
[Q&A] update

mux99 committed:
[supported] start of list

mux99 committed:
[debugging] TODO

mux99 committed:
update ss command

mux99 committed:
add mptcp_info exemple

mux99 committed:
update landing

added explanations of what MPTCP is

mux99 committed:
commented unfinished parts

mux99 committed:
minor fixes

mux99 committed:
fix redundancy

matttbe commented on datails.md:
```diff
@@ -0,0 +1,9 @@
+---
+layout: home
+title: Inner workigns
```
"Implementation details" maybe?

matttbe commented on debugging.md:
```diff
@@ -0,0 +1,16 @@
+<!-- ## nstat
+https://linux.die.net/man/8/nstat
+
+==TODO==
```
Can you clarify the TODO (in the code, not here)?
Does it mean:
- you or Dorian will do it later
- you don't know what kind of content we want here, you need some help for you doing that later
- you both never used it, and didn't try, or it looks too complex, someone else needs to do that

Same for all the other TODOs here

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+	__u64	mptcpi_bytes_sent;
+	__u64	mptcpi_bytes_received;
+	__u64	mptcpi_bytes_acked;
+    __u8    mptcpi_subflows_total;
```
not aligned.

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+struct mptcp_info info;
+socklen_t info_len = sizeof(struct mptcp_info);
+
+getsockopt(r->con->fd, SOL_MPTCP, MPTCP_INFO, &info, &info_len)
```
use `fd`, check the returned value, add `;`

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+
+getsockopt(r->con->fd, SOL_MPTCP, MPTCP_INFO, &info, &info_len)
+
+if ((int)offsetof(struct mptcp_info, mptcpi_subflows_total)-(int)info_len < 0){
```
a comment is needed here above to understand what you are trying to do.

spaces around `-`

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+getsockopt(r->con->fd, SOL_MPTCP, MPTCP_INFO, &info, &info_len)
+
+if ((int)offsetof(struct mptcp_info, mptcpi_subflows_total)-(int)info_len < 0){
+    console.log("%b",(uintmax_t)info.mptcpi_subflows_total);
```
use `printf()`
Also, `%b`?
And a cast to `uintmax_t`?

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+
+if ((int)offsetof(struct mptcp_info, mptcpi_subflows_total)-(int)info_len < 0){
+    console.log("%b",(uintmax_t)info.mptcpi_subflows_total);
+}
```
we need an example for `FULLINFO`, that's the complex one.

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+**note: depending on the version of MPTCP used, some options might not be present**
+the presence or not of an option can be checked in two ways:
+- first, if the variable used to store the data is initialized at zero, missing values will be zero.
+- second, the relative position of an entry in the structure can be compared to the `optlen` value using the `offsetof` function.
```
maybe best to put that above

matttbe commented on implementation.md:
```diff
@@ -0,0 +1,172 @@
+	- `mptcpi_subflows_total`:
+	- `num_subflows`:
+
+- **==TODO==** -->
```
Same here, TODO for who?

matttbe commented on index.md:
```diff
@@ -58,4 +62,8 @@ for more details.
+
+<!-- This site is new and still evolving, so please refer to the [Linux MPTCP Upstream Project wiki](https://github.com/multipath-tcp/mptcp_net-next/wiki) for additional information.
+
+_For out-of-tree kernels before v5.6 and an implementation of the experimental [MPTCP v0](https://www.rfc-editor.org/rfc/rfc6824.html) protocol, see https://multipath-tcp.org/_ -->
```
If you comment, please explain why

matttbe commented on installation.md:
```diff
@@ -0,0 +1,144 @@
+
+    `mptcpize run <app command>`  
+    this works by modifying the behaviour of the underlaying lib-c  
+    *note: this does does not always work and should not be used for production*
```
it can be used in production, it is just that some sysadmin doesn't like changing the behaviour of the libC. Like other techniques below, the best is to modify the app to natively support MPTCP where required.

matttbe commented on installation.md:
```diff
@@ -0,0 +1,144 @@
+- [systemtap](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/getting-started-with-multipath-tcp_configuring-and-managing-networking#preparing-rhel-to-enable-mptcp-support_getting-started-with-multipath-tcp)
+
+## ss commands
+the `ss` command on linux systems has an option to list MPTCP socket `-M`. the recommanded call is `ss -Menita`
```
a bit all over, there are typos, capital letter are missing at the beginning of a sentence, etc. That's not pro. Can you fix that please?

For grammar checks, you can be assisted with some tools, e.g. https://languagetool.org

matttbe commented on installation.md:
```diff
@@ -0,0 +1,144 @@
+the `ss` command on linux systems has an option to list MPTCP socket `-M`. the recommanded call is `ss -Menita`
+
+- `M`, MPTCP sockets
+- `t`, TCP including subflows created by MPTCP
```
maybe best to remove `-M` (and `-e`) for the recommended command: so only to list MPTCP connections.

Then below, mention a second command with `-t` which will list TCP connections, including MPTCP subflow (if executed with root, we should see `tcp-ulp-mptcp`)

mux99 replied:
you mean remove `-t`and put it in another command?

matttbe commented on installation.md:
```diff
@@ -0,0 +1,144 @@
+[man page](https://www.commandlinux.com/man-page/man8/ss.8.html)
+
+
+## Making use of multiple streams
```
streams? do you mean subflows or paths? We should probably use `subflows` but explain it before

matttbe commented on installation.md:
```diff
@@ -0,0 +1,144 @@
+
+
+## Making use of multiple streams
+For the apps to be able to use multiple streams the kernel needs to be told what interfaces can be used to do so.
```
we need to explain two main things:
- endpoints needs to be set `ip mptcp`: why + how
- routing needs to be configured `ip route` + `ip rule`: why + how
(see my messages on the chat for the why)

Then we can say that some tools can help to do this configuration: Network Manager 1.40+, and mptcpd, with link and maybe explanations.

Please at least add the structure and info you have

mux99 replied:
> Please at least add the structure and info you have

can you elaborate? I don't get it

matttbe commented on installation.md:
```diff
@@ -0,0 +1,144 @@
+</details>
+
+<details markdown="block">
+<summary>Automatic Configuration</summary>
```
these are all old workarounds, can be removed

matttbe commented on questions.md:
```diff
@@ -0,0 +1,34 @@
+---
+
+## Are there any security & privacy concerns?
+MPTCP aims to maintain the same level of security as traditional TCP, with specific mechanisms to counter common network attacks. While designed to be secure, vulnerabilities, as with any protocol, can occur. But the Linux kernel is also known to quickly fix security issues once they have been identified.
```
globally, not just here, can you go to the next line after ~80 or 100 chars? That's not just for the style, that's because it is easier to review, and see the modifications later.

matttbe commented on supported.md:
```diff
@@ -0,0 +1,10 @@
+---
+the apps listed bellow support MPTCP natively, other apps can be forced to use it by following istructions [here](installation.html#force-mptcp)
+
+| Name | PR | default? | how to use |
```
I would use `Version` instead of `PR`: to know from which version it is supported.

(and capital letters)

mux99 replied:
do we keep the link to the PR on the version number?

matttbe replied:
yes, good idea

matttbe commented on supported.md:
```diff
@@ -0,0 +1,10 @@
+title: Supported apps
+nav_order: 7
+---
+the apps listed bellow support MPTCP natively, other apps can be forced to use it by following istructions [here](installation.html#force-mptcp)
```
probably best to explain  what "natively" means.

matttbe commented on supported.md:
```diff
@@ -0,0 +1,10 @@
+
+| Name | PR | default? | how to use |
+| --- | --- | --- | --- |
+| [Lighttpd 1.4](https://www.lighttpd.net/) | [github](https://github.com/lighttpd/lighttpd1.4/pull/132) | no | `server.feature-flags = ( "server.network-mptcp" => "enable" )` |
```
Add something: please contact us if you know other apps supporting MPTCP natively.

mux99 replied:
what contact do we recommend here?

matttbe replied:
> Please open a new [issue](https://github.com/multipath-tcp/mptcp.dev/issues) or a [pull request](https://github.com/multipath-tcp/mptcp.dev/pulls) if you know other apps supporting MPTCP natively.

mux99 committed:
fixes - first round

mux99 committed:
fix spacing

mux99 committed:
fix revert code spacing

mux99 committed:
fixes

mux99 committed:
fix

mux99 committed:
fix footer + browser description

mux99 committed:
Update _config.yml

mux99 committed:
Update index.md

mux99 committed:
Update _config.yml

CrapsDorian committed:
add questions from cURL

mux99 committed:
mptcp_full_info

mux99 committed:
fixes

mux99 committed:
fix schema background color

mux99 committed:
fix class not working when deployed

mux99 committed:
add kramdown functionality

mux99 committed:
fixes

CrapsDorian committed:
add row Making Use of Multiple Streams version

mux99 committed:
better details

mux99 committed:
new class

mux99 committed:
move link and add subflow == path

mux99 committed:
add favicon

mux99 committed:
links compliance

mux99 committed:
fix, favicon

mux99 committed:
fix

mux99 committed:
update "making use of multiple streams"

mux99 committed:
add link

mux99 committed:
q&a update

CrapsDorian committed:
add dark theme

CrapsDorian committed:
fix dark theme

CrapsDorian committed:
fix dark theme

mux99 committed:
fixes + new question in q&a

mux99 committed:
fix attempt
